server/apiv2/handlers/tso_keyspace_group.go

```diff
@@ -183,6 +190,48 @@ func FinishSplitKeyspaceByID(c *gin.Context) {
 	c.JSON(http.StatusOK, nil)
 }
 
+// AllocNodeForKeyspaceGroupParams defines the params for allocating nodes for keyspace groups.
+type AllocNodeForKeyspaceGroupParams struct {
+	Replica int `json:"replica"`
+}
+
+// AllocNodeForKeyspaceGroup allocates nodes for keyspace group.
+func AllocNodeForKeyspaceGroup(c *gin.Context) {
+	id, err := validateKeyspaceGroupID(c)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, "invalid keyspace group id")
+		return
+	}
+	svr := c.MustGet(middlewares.ServerContextKey).(*server.Server)
+	manager := svr.GetKeyspaceGroupManager()
+	allocParams := &AllocNodeForKeyspaceGroupParams{}
+	err = c.BindJSON(allocParams)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrBindJSON.Wrap(err).GenWithStackByCause())
+		return
+	}
+	if manager.GetNodesNum() < allocParams.Replica || allocParams.Replica < 1 {
```

Comment:
Assume that the group has a specific user kind, here we just GetNodesNum to do the check. Does it make sense?

Reply:
https://github.com/tikv/pd/pull/6274#discussion_r1160380892